File: app/src/main/java/com/toly1994/ivideo/presenter/HomePresenter.java
package com.toly1994.ivideo.presenter;

import android.content.Context;
import com.toly1994.ivideo.model.VideoInfo;
import com.toly1994.ivideo.model.VideoScanner;
import com.toly1994.ivideo.view.IView;

import java.util.List;

/**
 * 作者：张风捷特烈<br/>
 * 时间：2019/4/1/001:19:33<br/>
 * 邮箱：1981462002@qq.com<br/>
 * 说明：
 */
public class HomePresenter implements IHomePresenter {
    private IView mIView;
    private final VideoScanner mVideoScanner;
    private List<VideoInfo> videos;

    public HomePresenter(IView view) {
        mIView = view;
        mVideoScanner = new VideoScanner();
    }

    @Override
    public void render(boolean force) {
        if (!force && videos != null) {//非强制加载,且视频已加载
            mIView.render(videos);//使用加载过的视频
            return;
        }

        mVideoScanner.loadVideo((Context) mIView, videos -> {
            this.videos = videos;
            mIView.render(videos);
        }, () -> {
                mIView.render(this.videos);
        });
    }

    @Override
    public void delete(Context context, int videoId) {
        mVideoScanner.delete(context, videoId);
        render(true);
    }
}

File: app/src/main/java/com/toly1994/ivideo/presenter/home/HomePresenter.kt
package com.toly1994.ivideo.presenter.home

import android.content.Context
import com.toly1994.ivideo.app.Cons
import com.toly1994.ivideo.app.api.VideoSourceApiImpl
import com.toly1994.ivideo.app.utils.SpUtils
import com.toly1994.ivideo.view.home.IHomeView

/**
 * 作者：张风捷特烈<br></br>
 * 时间：2019/4/1/001:19:33<br></br>
 * 邮箱：1981462002@qq.com<br></br>
 * 说明：
 */
class HomePresenter(private val mIHomeView: IHomeView) : IHomePresenter {

    companion object {
        var videoSourceApi: VideoSourceApiImpl = VideoSourceApiImpl()
    }

    override fun render(force: Boolean) {
        val isListDir = SpUtils.newInstance().getBoolean(Cons.SP_LIST_DIR)//是否列表文件夹

        mIHomeView.showLoading()

        videoSourceApi.getVideos(mIHomeView.context, force, { videos ->
            show(isListDir)
        }, { show(isListDir) })
    }

    private fun show(isListDir: Boolean) {
        if (isListDir) {
            mIHomeView.renderDir(videoSourceApi.videoDirs)
        } else {
            mIHomeView.renderVideo(videoSourceApi.videos)
        }
        mIHomeView.hideLoading()
    }

    override fun delete(context: Context, videoId: Int) {
        videoSourceApi.delete(context, videoId)
        render(true)
    }

    override fun filter(name: String) {
        mIHomeView.renderVideo(videoSourceApi.filterName(name))
    }

    fun isLogin() {

    }

}
